machine_translation/sampling.py

```diff
@@ -90,15 +83,17 @@ def do(self, which_callback, *args):
         #  can be different
         batch = args[0]
 
-        # TODO: this is problematic for boundary conditions, eg. last batch
         sample_idx = numpy.random.choice(
             self.config['batch_size'], self.config['hook_samples'],
             replace=False)
         src_batch = batch[self.main_loop.data_stream.mask_sources[0]]
         trg_batch = batch[self.main_loop.data_stream.mask_sources[1]]
 
-        input_ = src_batch[sample_idx, :]
-        target_ = trg_batch[sample_idx, :]
+        try:
+            input_ = src_batch[sample_idx, :]
+            target_ = trg_batch[sample_idx, :]
+        except IndexError:
+            return
 
```

Comment:
please see [this](https://github.com/orhanf/blocks-examples/commit/445e78114008165f4f38467e430354bae55b4080) commit which gets the batch size accordingly.